src/main/java/digital/pragmatech/testing/diagnostic/ContextDiagnosticApplicationInitializer.java

```diff
@@ -30,4 +30,16 @@ public void initialize(ConfigurableApplicationContext applicationContext) {
           }
         });
   }
+
+  @Override
+  public boolean equals(Object that) {
```

Comment:
alternatively, this could be implemented as
```
if (!applicationContext.getBeanFactory().containsSingleton(BEAN_CONTEXT_DIAGNOSTIC)) {
  applicationContext
      .getBeanFactory()
      .registerSingleton(BEAN_CONTEXT_DIAGNOSTIC, completedDiagnostic);
}
```
but this solution with `equals` looks as a proper solution